Create CCA Evidence (#5)
veraison/ccatoken: README.md

# ccatoken

> This repository is an implementation of Confidential Computing Architecture (CCA) Attestation Token Library.

Reviewer: ```suggestion
This repository is an implementation of the Confidential Computing Architecture (CCA) Attestation Token.
```

> This work is based on the Attestation Token as detailed in Realm Management Monitor Specificiation [RMM Spec](https://developer.arm.com/documentation/den0137/a/?lang=en)

Reviewer: ```suggestion
This work is based on the Attestation Token as detailed by the [Realm Management Monitor (RMM) Specification](https://developer.arm.com/documentation/den0137/latest)
```

> The package exposes the following functionalities:

Reviewer: ```suggestion
The package allows to:
```

> * Construct CCA Evidence by setting the CCA platform and realm claims.

Reviewer: ```suggestion
* Construct CCA Evidence by separately setting CCA platform and Realm claims.
```

> * Sign the claims to get the CCA token (as CBOR bytes)

Reviewer: ```suggestion
* Sign and serialise the CCA token to CBOR
```

> * Get the CCA platform and realm claims from received CCA token

Reviewer: ```suggestion
* Decode a CBOR-encoded CCA token
* Extract the CCA platform and Realm claims
```
